.github/workflows/update-gardenlogin.yaml

```diff
@@ -2,63 +2,153 @@ name: gardenlogin-updater
 
 permissions:
   contents: write
-  actions: read
-  repository-projects: write
+  actions:  read
 
 on:
   release:
-    types:
-      - published
+    types: [published]
+
+  # Manual / ad‑hoc trigger
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag or version to build & dispatch (e.g. v1.2.3)'
+        required: true
+        type: string
+      skip_build:
+        description: "Skip build & upload; send dummy checksums"
+        required: false
+        type: boolean
+        default: true
+      dry_run:
+        description: "Dry run: disable external actions (no uploads/dispatches)"
+        required: false
+        type: boolean
+        default: false
+
 jobs:
   update_gardenlogin_in_homebrew_tap_and_chocolatey_packages:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+      cancel-in-progress: true
+
+    # ───── Single source of truth for the tag (expression evaluated outside the shell) ─────
+    env:
+      TAG: ${{ inputs.tag || github.ref_name }}
+
     steps:
-      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
-      - uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # v5.5.0
+      # ───── Checkout ─────────────────────────────────────────────────────────
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683  # v4.2.2
         with:
-          go-version: '1.24.4'
-      - name: Build the binary-files
-        id: build_binary_files
+          fetch-depth: 1
```

Comment:
you have to specify the `ref` (the tag) to checkout. On publish event this is probably not so relevant (but makes it explicit), but when triggered manually, probably the wrong ref is checked out.